configure.ac

```diff
@@ -1531,12 +1531,14 @@ if test "${enable_exporting_kinesis}" = "yes" -o \
     enable_cxx_linker="yes"
 fi
 
-# Unconditionally link with the libatomic since:
+# Unconditionally link with the libatomic except on macos since:
 # - if atomics are not needed, the library will be unused
 # - if atomics are needed, either atomics are fulfilled by the compiler builtins
 #   and the library is unused or the library is needed anyway.
-OPTIONAL_ATOMIC_LIBS="-latomic"
-AC_SUBST([OPTIONAL_ATOMIC_LIBS])
+if test "${build_target}" != "macos"; then
```

Comment:
Should this be based on `AC_CHECK_LIB` instead of `build_target`?

Reply:
I don't think there's even libatomic in MacOS. Searched brew, there's only libatomic_ops, but not sure if it's the same thing....

Reply:
yes but wouldnt that make the fix universal for all various random places where libatomic is not available. Maybe 2nd hand apple logo branded computers are not the only such system in the world.

Reply:
I think the whole point of #12366 was to link the library always, and let the compiler decide. Most likely we would have some trouble already if it caused compiling issues on other platforms too...

Reply:
well we can merge it as it is as long as we take care of this after the release

Reply:
I can have a go at making this work right, but not sure if I can make it today...